pkg/container-hook/tracer.go

```diff
@@ -397,22 +414,31 @@ func (n *ContainerNotifier) watchPidFileIterate(
 	}
 
 	// This unblocks whoever is accessing the pidfile
-	defer pidFileDirNotify.ResponseAllow(data)
+	defer n.pidFileDirNotify.ResponseAllow(data)
 
 	path, err := data.GetPath()
 	if err != nil {
 		return false, err
 	}
 	path = filepath.Join(host.HostRoot, path)
 
-	// Consider files identical if they have the same device/inode,
-	// even if the paths differ due to symlinks (for example,
-	// the event's path is /run/... but the runc --pid-file argument
-	// uses /var/run/..., where /var/run is a symlink to /run).
-	filesAreIdentical, err := checkFilesAreIdentical(path, pidFile)
-	if err != nil {
-		return false, err
-	} else if !filesAreIdentical {
+	n.pendingMu.Lock()
+	var pc *pendingContainer = nil
+	for pidFile := range n.pendingContainers {
+		// Consider files identical if they have the same device/inode,
+		// even if the paths differ due to symlinks (for example,
+		// the event's path is /run/... but the runc --pid-file argument
+		// uses /var/run/..., where /var/run is a symlink to /run).
+		filesAreIdentical, err := checkFilesAreIdentical(path, pidFile)
+		if err == nil && filesAreIdentical {
```

Comment:
I think not: if the file does not exist, it might be that the current iteration of the loop is a container with a problem but we should not stop the fanotify monitoring because we use a single fanotify fd for all containers now.